yt/frontends/stream/definitions.py

```diff
@@ -177,7 +180,9 @@ def process_data(data, grid_dims=None):
     # At this point, we have arrays for all our fields
     new_data = {}
     for field in data:
-        n_shape = len(data[field].shape)
+        n_shape = 3
```

Comment:
I wasn't incredibly happy with this.  Right now, if it's callable, it assumes it's 3D.  I'm wondering if maybe we could have a decorator that supplied some metadata about the field that would either be returned or would be assigned to the function object.

Reply:
we could provide a `@shape(...)` decorator that would add a `.shape` attribute to callable field functions so this line would work transparently without change. It would work something like this

```python
@shape(1, 2, 3)
def my_callable_field(...):
    ....

my_callable_field.shape  == (1, 2, 3)
```

that said, if we only really car about dimensionality, then `len(data[field].shape)`, then we should just call `data[field].ndim` and have a corresponding, simpler decorator instead.

```python
@ndim(3)
def my_callable_field(...):
    ...
```

Reply:
Great idea.  I personally think an `ndim` would be minimum ... I think a more expansive one could come later.